src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -32,8 +32,30 @@
 namespace ArborX
 {
 
+struct PairIndexRank
+{
+  int index;
+  int rank;
+  friend bool operator==(ArborX::PairIndexRank lhs, ArborX::PairIndexRank rhs)
+  {
+    return lhs.index == rhs.index && lhs.rank == rhs.rank;
+  }
+  friend bool operator<(ArborX::PairIndexRank lhs, ArborX::PairIndexRank rhs)
+  {
+    return lhs.index < rhs.index ||
+           (lhs.index == rhs.index && lhs.rank < rhs.rank);
+  }
+  friend std::ostream &operator<<(std::ostream &stream,
+                                  ArborX::PairIndexRank const &pair)
+  {
+    return stream << '[' << pair.index << ',' << pair.rank << ']';
+  }
```

Comment:
```suggestion
```
No we don't want to provide this as part of ArborX

Reply:
It is required for testing, though. I'm not sure there is an easy way to avoid it.
```
In file included from <snip>arborx/test/tstDistributedTree.cpp:24:
In file included from <snip>/arborx/test/Search_UnitTestHelpers.hpp:16:
<snip>/arborx/test/boost_ext/KokkosPairComparison.hpp:33:13:
error: invalid operands to binary expression ('basic_ostream<char, std::char_traits<char>>' and
'const Kokkos::pair<ArborX::PairIndexRank, float>::first_type' (aka 'const ArborX::PairIndexRank'))
  os << '(' << p.first << ',' << p.second << ')';
```

Reply:
Yes I know but we would define it in the tests source files, not in the library.

Reply:
But the declaration would still need to be a part of the class, right, as it has to be a friend? I tried to do a quick patch:
```diff
--- a/src/details/ArborX_DetailsDistributedTreeImpl.hpp
+++ b/src/details/ArborX_DetailsDistributedTreeImpl.hpp
@@ -36,20 +36,15 @@ struct PairIndexRank
     return lhs.index < rhs.index ||
            (lhs.index == rhs.index && lhs.rank < rhs.rank);
   }
-  friend std::ostream &operator<<(std::ostream &stream,
-                                  ArborX::PairIndexRank const &pair)
-  {
-    return stream << '[' << pair.index << ',' << pair.rank << ']';
-  }
 };

 namespace Details
diff --git a/test/ArborX_BoostRTreeHelpers.hpp b/test/ArborX_BoostRTreeHelpers.hpp
index d37d9b19..64d42c7c 100644
--- a/test/ArborX_BoostRTreeHelpers.hpp
+++ b/test/ArborX_BoostRTreeHelpers.hpp
@@ -23,6 +23,14 @@
#ifdef ARBORX_ENABLE_MPI
#include <ArborX_DistributedTree.hpp> // PairIndexRank
+#include <ArborX_DistributedTree.hpp> // PairIndexRank
+std::ostream &operator<<(std::ostream &stream,
+                         ArborX::PairIndexRank const &pair)
+{
+  return stream << '[' << pair.index << ',' << pair.rank << ']';
+}
#endif
 
 namespace BoostRTreeHelpers
 {
 // NOTE: The balancing algorithm does not really matter since the tree is
```
but that did not work:
```
<snip>/boost/test/tools/detail/print_helper.hpp:53:13: error: static_assert failed due to requirement 
'boost::has_left_shift<std::ostream, ArborX::PairIndexRank, boost::binary_op_detail::dont_care>::value' "Type
has to implement operator<< to be printable"
            BOOST_STATIC_ASSERT_MSG( (boost::has_left_shift<std::ostream,T>::value),
            ^                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
It is possible I'm doing something wrong.

Reply:
Define it in the `ArborX` namespace

Reply:
We own the namespace so we are allowed to do it but Boost.Test supports other mechanisms to print objects.